scripts/setupPythonEnvironment.bash

```diff
@@ -91,84 +85,47 @@ then
     then
         echo "If GEOSX is configured to use pygeosx, you may need to run \"make pygeosx\""
         echo "before setting up the geosx_python_tools!"
-    else
-        echo "Note: if you use the \"-m\" argument, this script will try to build a version of miniconda"
-        echo "that can support the target tools"
     fi
     exit 1
 fi
 
 
-# If a virtual environment is not explicitly requested,
-# try installing packages directly
-if ! $INSTALL_VIRTUAL
-then
-    echo "Attempting to install packages directly in target python environment..."
-    for p in "${TARGET_PACKAGES[@]}"
-    do
-        if [ -d "$p" ]
+# Install packages
+echo "Installing python packages..."
+for p in "${TARGET_PACKAGES[@]}"
+do
+    if [ -d "$p" ]
+    then
+        echo "  $p"
+        if $VERBOSE
+            INSTALL_MSG=$($PYTHON_TARGET -m $PIP_CMD install $p)
+            INSTALL_RC=$?
         then
-            echo "  $p"
-            RES=$($PYTHON_TARGET -m pip install $p 2>&1)
-            if [[ $RES =~ "Error" ]]
-            then
-                echo "  (cannot install target packes directly)"
-                INSTALL_VIRTUAL=true
-                break
-            fi
-        else
-            echo "Could not find target package: $p"
+            INSTALL_MSG=$($PYTHON_TARGET -m $PIP_CMD install $p 2>&1)
+            INSTALL_RC=$?
```

Comment:
Maybe get `INSTALL_RC=$?` out of the `if`/`then`?
As you wish, having the error code the line just after is also good!

Reply:
I did try that, but apparently the op code gets reset when it exits the if block.  Thanks for your suggestions!